android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/BitmapExtension.kt

```diff
@@ -30,3 +31,8 @@ fun Bitmap.encodeToByteArray(): ByteArray {
 fun ByteArray.decodeToBitmap(offset: Int = 0): Bitmap {
   return BitmapFactory.decodeByteArray(this, offset, this.size)
 }
+
+fun String.base64toBitmap(offset: Int = 0): Bitmap {
+  val decodedBytes = Base64.decode(this, Base64.DEFAULT)
+  return BitmapFactory.decodeByteArray(decodedBytes, offset, decodedBytes.size)
+}
```

Comment:
If you re-use this functionality
```
  viewModelScope.launch(dispatcherProvider.io()) {
          registerRepository.loadResource<Binary>(resourceId.extractLogicalIdUuid())?.let { binary ->
            it.menuIconConfig!!.decodedBitmap = binary.data.decodeToBitmap()
          }
        }
```
The extension will no longer be necessary. Where `resourceId` is the image configuration reference.